specification/appendix-d-observability.md

```diff
@@ -51,6 +51,74 @@ The following describes how keys in [flag metadata](types.md#flag-metadata) are
 | ---------------------------- | ----------------------- | ----------------- | -------- | ------------------------------------------------------------------------------------------------ |
 | `feature_flag.provider.name` | `name`                  | `Recommended`     | `string` | The name of the provider as defined in the `provider metadata`, available in the `hook context`. |
 
+## Telemetry Hook Implementation Guide
+
+This section provides guidance for implementing observability hooks that emit OpenTelemetry signals during feature flag evaluations. The recommendations ensure consistency across SDK implementations while allowing for language-specific idioms.
+
+### Signal Emission Patterns
+
+Telemetry hooks can emit OpenTelemetry signals in three distinct ways:
+
+| Pattern              | Recommended | Advantages                                                                                                                                                         | Disadvantages                                                                                                                                          |
+| -------------------- | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| **Span Events**      | ✓           | Leverages existing trace configuration and tooling. Minimal overhead - no additional spans created. Maintains trace context relationships. Simpler than creating spans. | Requires an active span to function. Must gracefully handle absence of active span. Limited to span lifetime and context.                              |
+| **Event Logging**    |             | Works independently without active spans. Aligns with OpenTelemetry's emerging direction. Suitable for environments without tracing. Simpler implementation model. | No automatic trace correlation. May require separate correlation mechanisms. Event logging standards still evolving.                                   |
+| **Standalone Spans** |             | Complete distributed tracing per evaluation. Detailed timing information. Full span lifecycle control. Rich performance metrics.                                   | Creates one span per evaluation. May clutter trace visualizations. Increased overhead and resource usage. Potential performance impact at scale. More complex implementation. |
+
+> [!NOTE]
+> While span events are recommended for their low overhead and ease of use, OpenTelemetry is trending toward using log-based events instead of span events. SDK implementations may choose to prioritize event logging as standards evolve.
```

Comment:
Relevant, but I don't know if that should be linked to here:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/oteps/4430-span-event-api-deprecation-plan.md